homeassistant/components/sensibo/number.py

```diff
@@ -0,0 +1,130 @@
+"""Number platform for Sensibo integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import async_timeout
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, LOGGER, SENSIBO_ERRORS, TIMEOUT
+from .coordinator import SensiboDataUpdateCoordinator
+
+
+@dataclass
+class SensiboEntityDescriptionMixin:
+    """Mixin values for Sensibo entities."""
+
+    remote_key: str
+
+
+@dataclass
+class SensiboNumberEntityDescription(
+    NumberEntityDescription, SensiboEntityDescriptionMixin
+):
+    """Class describing Sensibo Number entities."""
+
+
+NUMBER_TYPES = (
+    SensiboNumberEntityDescription(
+        key="calibration_temp",
+        remote_key="temperature",
+        name="Temperature calibration",
+        icon="mdi:thermometer",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+        min_value=-10,
+        max_value=10,
+        step=0.1,
+    ),
+    SensiboNumberEntityDescription(
+        key="calibration_hum",
+        remote_key="humidity",
+        name="Humidity calibration",
+        icon="mdi:water",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+        min_value=-10,
+        max_value=10,
+        step=0.1,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sensibo number platform."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        SensiboNumber(coordinator, device_id, description)
+        for device_id, device_data in coordinator.data.items()
+        for description in NUMBER_TYPES
+        if device_data["hvac_modes"] and device_data["temp"]
+    )
+
+
+class SensiboNumber(CoordinatorEntity, NumberEntity):
+    """Representation of a Sensibo numbers."""
+
+    coordinator: SensiboDataUpdateCoordinator
+    entity_description: SensiboNumberEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SensiboDataUpdateCoordinator,
+        device_id: str,
+        entity_description: SensiboNumberEntityDescription,
+    ) -> None:
+        """Initiate Sensibo Number."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._device_id = device_id
+        self._client = coordinator.client
+        self._attr_unique_id = f"{device_id} {entity_description.key}"
```

Comment:
Last comment, according to https://developers.home-assistant.io/docs/entity_registry_index/#unique-id unique id should be formatted as `{unique_id}-{sensor_type}`.
@bdraco is this correct? should it changed to `f"{device_id}-{entity_description.key}"`?

Reply:
I have seen everything from space, comma, dash etc. Happy to confirm to "-" if that's what the documentation provides as example.

Reply:
Let's see what @bdraco has to say, just to be aligned with recommended guidelines as changing this later is complicated

Reply:
`-` is preferred in new code. There isn't much value in changing existing code though as it carries the risk of migration.